CMakeLists.txt

```diff
@@ -0,0 +1,117 @@
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    RESULT_VARIABLE OPENSSL_CONFIGURE_RESULT
+  )
+  if(NOT (${OPENSSL_CONFIGURE_RESULT} EQUAL "0"))
+    message(FATAL_ERROR "Failed to configure OpenSSL with error: ${OPENSSL_CONFIGURE_RESULT}")
+  endif()
+endif()
+
+# Include the generated CMake config.
+include(openssl-config.cmake)
+
+# Add static library targets.
+foreach(static_target ${OPENSSL_STATIC_TARGETS})
+  add_library(openssl_${static_target} STATIC ${OPENSSL_SOURCES_${static_target}})
+  target_include_directories(openssl_${static_target} PRIVATE ${OPENSSL_INCLUDES_${static_target}})
+  target_compile_definitions(openssl_${static_target} PRIVATE ${OPENSSL_DEFINES_${static_target}})
+  target_link_libraries(openssl_${static_target} PRIVATE ${OPENSSL_LINK_LIBRARIES_${static_target}})
+  set_target_properties(openssl_${static_target} PROPERTIES OUTPUT_NAME "${static_target}")
+  set_target_properties(openssl_${static_target} PROPERTIES PREFIX "") # prevent lib prefix
+  if(OPENSSL_POSITION_INDEPENDENT_CODE)
+    set_property(TARGET openssl_${static_target} PROPERTY POSITION_INDEPENDENT_CODE ON)
+  endif()
+  if(OPENSSL_DEPENDENCIES_${static_target})
+    add_dependencies(openssl_${static_target} ${OPENSSL_DEPENDENCIES_${static_target}})
+  endif()
+  add_library(OpenSSL::${static_target} ALIAS openssl_${static_target})
+endforeach()
+
+# Add shared library targets.
+foreach(shared_target ${OPENSSL_SHARED_TARGETS})
+  add_library(openssl_${shared_target} SHARED ${OPENSSL_SOURCES_${shared_target}} ${OPENSSL_LINK_DEFS_${shared_target}})
+  target_include_directories(openssl_${shared_target} PRIVATE ${OPENSSL_INCLUDES_${shared_target}})
+  target_compile_definitions(openssl_${shared_target} PRIVATE ${OPENSSL_DEFINES_${shared_target}})
+  target_link_libraries(openssl_${shared_target} PRIVATE ${OPENSSL_LINK_LIBRARIES_${shared_target}})
+  set_target_properties(openssl_${shared_target} PROPERTIES OUTPUT_NAME "${shared_target}")
+  set_target_properties(openssl_${shared_target} PROPERTIES PREFIX "") # prevent lib prefix
+  if(OPENSSL_POSITION_INDEPENDENT_CODE)
+    set_property(TARGET openssl_${shared_target} PROPERTY POSITION_INDEPENDENT_CODE ON)
+  endif()
+  if(OPENSSL_HAS_VERSION_SCRIPT_SUPPORT)
+    foreach(lddef ${OPENSSL_LINK_DEFS_${shared_target}})
+      set_property(TARGET openssl_${shared_target} APPEND_STRING
+                    PROPERTY LINK_FLAGS " -Wl,--version-script=${lddef}")
+      set_target_properties(openssl_${shared_target} PROPERTIES LINK_DEPENDS ${lddef})
+    endforeach()
+  endif()
+  if(OPENSSL_DEPENDENCIES_${shared_target})
+    add_dependencies(openssl_${shared_target} ${OPENSSL_DEPENDENCIES_${shared_target}})
+  endif()
+  add_library(OpenSSL::${shared_target} ALIAS openssl_${shared_target})
+endforeach()
+
+# Add DSO module targets.
+foreach(dso_target ${OPENSSL_DSO_TARGETS})
+  add_library(openssl_${dso_target} SHARED ${OPENSSL_SOURCES_${dso_target}} ${OPENSSL_LINK_DEFS_${dso_target}})
```

Comment:
Actually, CMake has a different keyword for these: MODULE. You should use that instead of SHARED